instagram/src/components/Authentication/Authenticate.js

```diff
@@ -0,0 +1,25 @@
+import React from "react";
+// import LoginPage from "../LoginPage/Login";
+// import App from "../../App/App";
+
+const authenticate = App =>
+  class App extends React.Component {
+    constructor() {
+      super();
+      this.state = { loggedIn: false };
+    }
+    // componentDidMount() {
+    //   if (localStorage.getItem("username")) {
```

Comment:
(I know you have this commented out for a reason) But remember, that loggedIn is part of state and when you change the state you should call setState, but you're on the right track

Reply:
oh yeah, that's true, i need to change the state by using setState()..
i have commented it out because I can't get it to render either <App /> or <LoginPage /> still..